$OpenBSD: patch-src_libbluray_bluray_c,v 1.1 2022/02/27 12:43:24 sthen Exp $

- Fix dec_start() without menus
  8412dc1ee24b186a161902de1dd9f15f712041f5
- Drop "WARNING: BluRay profile 6 BD-J menu support is experimental"
  000af4c152f532d2347cb17df628592244a5f764

Index: src/libbluray/bluray.c
--- src/libbluray/bluray.c.orig
+++ src/libbluray/bluray.c
@@ -1132,12 +1132,13 @@ static void _fill_disc_info(BLURAY *bd, BD_ENC_INFO *e
         /* increase player profile and version when 3D or UHD disc is detected */
 
         if (index->indx_version >= ('0' << 24 | '3' << 16 | '0' << 8 | '0')) {
-            BD_DEBUG(DBG_CRIT | DBG_BLURAY, "WARNING: BluRay profile 6 BD-J menu support is experimental\n");
+            BD_DEBUG(DBG_BLURAY, "Detected 4K UltraHD (profile 6) disc\n");
             /* Switch to UHD profile */
             psr_init_UHD(bd->regs, 1);
         }
         if (((index->indx_version >> 16) & 0xff) == '2') {
             if (index->app_info.content_exist_flag) {
+                BD_DEBUG(DBG_BLURAY, "Detected Blu-Ray 3D (profile 5) disc\n");
                 /* Switch to 3D profile */
                 psr_init_3D(bd->regs, index->app_info.initial_output_mode_preference, 0);
             }
@@ -2419,6 +2420,11 @@ static int _add_known_playlist(BD_DISC *p, const char 
 
 static int _open_playlist(BLURAY *bd, const char *f_name, unsigned angle)
 {
+    if (!bd->title_list && bd->title_type == title_undef) {
+        BD_DEBUG(DBG_BLURAY | DBG_CRIT, "open_playlist(%s): bd_play() or bd_get_titles() not called\n", f_name);
+        disc_event(bd->disc, DISC_EVENT_START, bd->disc_info.num_titles);
+    }
+
     _close_playlist(bd);
 
     bd->title = nav_title_open(bd->disc, f_name, angle);
